src/app/adtwizardpage.h

```diff
@@ -0,0 +1,45 @@
+/***********************************************************************************************************************
+**
+** Copyright (C) 2023 BaseALT Ltd. <org@basealt.ru>
+**
+** This program is free software; you can redistribute it and/or
+** modify it under the terms of the GNU General Public License
+** as published by the Free Software Foundation; either version 2
+** of the License, or (at your option) any later version.
+**
+** This program is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with this program; if not, write to the Free Software
+** Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+**
+***********************************************************************************************************************/
+
+#ifndef ADTWIZARDPAGE_H
+#define ADTWIZARDPAGE_H
+
+#include "ui_checkwizardpage.h"
+
+template<typename TUi>
```

Comment:
It will be better to implement this class like this:
```
template<typename TUi>
class ADTWizardPage
{
public:
    ADTWizardPage();
    virtual ~ADTWizardPage();

private:
    TUi *ui;

public:
    TUi *operator->() { return ui; }
    TUi *operator->() const { return ui; }

private:
    ADTWizardPage(const ADTWizardPage &) = delete;
    ADTWizardPage(ADTWizardPage &&)      = delete;
    ADTWizardPage &operator=(const ADTWizardPage &) = delete;
    ADTWizardPage &operator=(ADTWizardPage &&) = delete;
};

#include "adtwizardpage.inl"
```

Reply:
Also please convert pointer to ADTWizardpage to simple field.

Reply:
fixed.